src/main.rs

```diff
@@ -20,37 +25,75 @@ use crate::hand_stats::HandStats;
 /// (2) print out the hand
 /// (3) compute and print the card stats
 fn main() {
-    let mut rng = rand::thread_rng();
-
-    println!("Sorted Deck:");
-    for id in 0..52 {
-        let card = Card { id };
-        if card.suit().id == 3 {
-            println!("  {}", card);
-        } else {
-            print!("  {}  ", card);
+    let args: PokerArgs = PokerArgs::parse();
+
+    // Matching the command
+    match &args.command {
+        // For draw-hand command
+        Some(ArgsCommands::DrawHands(DrawHand { hands_number })) => {
+            // If call draw-hands with hands_number as 5
+            if *hands_number == 5 {
+                output_specific::<5>(rand::thread_rng(), 20000);
+            }
+            // If call draw-hands with hands_number as 7
+            else if *hands_number == 7 {
+                output_specific::<7>(rand::thread_rng(), 20000);
+            }
+            // If hand_number is not 5 not 7
+            else {
+                println!("Invalid number. Enter 5 or 7")
+            }
+        }
+
+        // For statistics command
+        Some(ArgsCommands::Statistics(GetStatistics {
+            hands_number,
+            sample_number,
+        })) => {
+            if *hands_number == 5 {
+                output_specific::<5>(rand::thread_rng(), *sample_number);
+            } else if *hands_number == 7 {
+                output_specific::<7>(rand::thread_rng(), *sample_number);
+            } else {
+                println!("Invalid number. Enter 5 or 7")
+            }
+        }
+
+        None => {
+            // If neither sorted_deck option is called nor any other i.e. for cargo run command (Default run)
+            if !(args.sorted_deck) {
```

Comment:
Could we remove this case entirely, and let this be implicitly implemented by providing default arguments?   

Edit:  details in an earlier comment.